M/mlpack/build_tarballs.jl

```diff
@@ -0,0 +1,175 @@
+    LibraryProduct("libmlpack_julia_fastmks", :libmlpack_julia_fastmks),
+    LibraryProduct("libmlpack_julia_gmm_generate",
+        :libmlpack_julia_gmm_generate),
+    LibraryProduct("libmlpack_julia_gmm_probability",
+        :libmlpack_gmm_probability),
+    LibraryProduct("libmlpack_julia_gmm_train", :libmlpack_julia_gmm_train),
+    LibraryProduct("libmlpack_julia_hmm_generate",
+        :libmlpack_julia_hmm_generate),
+    LibraryProduct("libmlpack_julia_hmm_loglik", :libmlpack_julia_hmm_loglik),
+    LibraryProduct("libmlpack_julia_hmm_train", :libmlpack_julia_hmm_train),
+    LibraryProduct("libmlpack_julia_hmm_viterbi", :libmlpack_julia_hmm_viterbi),
+    LibraryProduct("libmlpack_julia_hoeffding_tree",
+        :libmlpack_julia_hoeffding_tree),
+    LibraryProduct("libmlpack_julia_kernel_pca", :libmlpack_julia_kernel_pca),
+    LibraryProduct("libmlpack_julia_kfn", :libmlpack_julia_kfn),
+    LibraryProduct("libmlpack_julia_kmeans", :libmlpack_julia_kmeans),
+    LibraryProduct("libmlpack_julia_knn", :libmlpack_julia_knn),
+    LibraryProduct("libmlpack_julia_krann", :libmlpack_julia_krann),
+    LibraryProduct("libmlpack_julia_lars", :libmlpack_julia_lars),
+    LibraryProduct("libmlpack_julia_linear_regression",
+        :libmlpack_julia_linear_regression),
+    LibraryProduct("libmlpack_julia_lmnn", :libmlpack_julia_lmnn),
+    LibraryProduct("libmlpack_julia_local_coordinate_coding",
+        :libmlpack_julia_local_coordinate_coding),
+    LibraryProduct("libmlpack_julia_lsh", :libmlpack_julia_lsh),
+    LibraryProduct("libmlpack_julia_mean_shift", :libmlpack_julia_mean_shift),
+    LibraryProduct("libmlpack_julia_nbc", :libmlpack_julia_nbc),
+    LibraryProduct("libmlpack_julia_nca", :libmlpack_julia_nca),
+    LibraryProduct("libmlpack_julia_nmf", :libmlpack_julia_nmf),
+    LibraryProduct("libmlpack_julia_pca", :libmlpack_julia_pca),
+    LibraryProduct("libmlpack_julia_perceptron", :libmlpack_julia_perceptron),
+    LibraryProduct("libmlpack_julia_preprocess_binarize",
+        :libmlpack_julia_preprocess_binarize),
+    LibraryProduct("libmlpack_julia_preprocess_describe",
+        :libmlpack_julia_preprocess_describe),
+    LibraryProduct("libmlpack_julia_preprocess_split",
+        :libmlpack_julia_preprocess_split),
+    LibraryProduct("libmlpack_julia_radical", :libmlpack_julia_radical),
+    LibraryProduct("libmlpack_julia_random_forest",
+        :libmlpack_julia_random_forest),
+    LibraryProduct("libmlpack_julia_softmax_regression",
+        :libmlpack_julia_softmax_regression),
+    LibraryProduct("libmlpack_julia_sparse_coding",
+        :libmlpack_julia_sparse_coding)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    "boost_jll",
+    "armadillo_jll",
+    "OpenBLAS_jll" # Is this necessary?
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
```

Comment:
For the last PowerPC error with libgfortran3
```
[17:30:48] /opt/powerpc64le-linux-gnu/powerpc64le-linux-gnu/sys-root//usr/local/lib/libopenblas64_.so.0: undefined reference to `GOMP_parallel@GOMP_4.0'
[17:30:48] collect2: error: ld returned 1 exit status
[17:30:48] make[2]: *** [src/mlpack/methods/decision_tree/CMakeFiles/generate_jl_decision_tree.dir/build.make:109: src/mlpack/bindings/julia/mlpack/build/bin/generate_jl_decision_tree] Error 1
[17:30:48] make[2]: Leaving directory '/workspace/srcdir/mlpack-3.3.0-a1/build'
[17:30:48] make[1]: *** [CMakeFiles/Makefile2:2154: src/mlpack/methods/decision_tree/CMakeFiles/generate_jl_decision_tree.dir/all] Error 2
[17:30:48] /opt/powerpc64le-linux-gnu/powerpc64le-linux-gnu/sys-root//usr/local/lib/libopenblas64_.so.0: undefined reference to `GOMP_parallel@GOMP_4.0'
[17:30:48] collect2: error: ld returned 1 exit status
[17:30:48] make[2]: *** [src/mlpack/methods/adaboost/CMakeFiles/generate_jl_adaboost.dir/build.make:109: src/mlpack/bindings/julia/mlpack/build/bin/generate_jl_adaboost] Error 1
[17:30:48] make[2]: Leaving directory '/workspace/srcdir/mlpack-3.3.0-a1/build'
[17:30:48] make[1]: *** [CMakeFiles/Makefile2:1221: src/mlpack/methods/adaboost/CMakeFiles/generate_jl_adaboost.dir/all] Error 2
[17:30:48] /opt/powerpc64le-linux-gnu/powerpc64le-linux-gnu/sys-root//usr/local/lib/libopenblas64_.so.0: undefined reference to `GOMP_parallel@GOMP_4.0'
[17:30:48] collect2: error: ld returned 1 exit status
[17:30:48] /opt/powerpc64le-linux-gnu/powerpc64le-linux-gnu/sys-root//usr/local/lib/libopenblas64_.so.0: undefined reference to `GOMP_parallel@GOMP_4.0'
[17:30:48] collect2: error: ld returned 1 exit status
[17:30:48] make[2]: *** [src/mlpack/methods/approx_kfn/CMakeFiles/generate_jl_approx_kfn.dir/build.make:109: src/mlpack/bindings/julia/mlpack/build/bin/generate_jl_approx_kfn] Error 1
[17:30:48] make[2]: Leaving directory '/workspace/srcdir/mlpack-3.3.0-a1/build'
[17:30:48] make[1]: *** [CMakeFiles/Makefile2:1655: src/mlpack/methods/approx_kfn/CMakeFiles/generate_jl_approx_kfn.dir/all] Error 2
[17:30:48] make[2]: *** [src/mlpack/methods/fastmks/CMakeFiles/generate_jl_fastmks.dir/build.make:109: src/mlpack/bindings/julia/mlpack/build/bin/generate_jl_fastmks] Error 1
[17:30:48] make[2]: Leaving directory '/workspace/srcdir/mlpack-3.3.0-a1/build'
[17:30:48] make[1]: *** [CMakeFiles/Makefile2:2439: src/mlpack/methods/fastmks/CMakeFiles/generate_jl_fastmks.dir/all] Error 2
[17:30:48] /opt/powerpc64le-linux-gnu/powerpc64le-linux-gnu/sys-root//usr/local/lib/libopenblas64_.so.0: undefined reference to `GOMP_parallel@GOMP_4.0'
[17:30:48] collect2: error: ld returned 1 exit status
[17:30:48] make[2]: *** [src/mlpack/methods/cf/CMakeFiles/generate_jl_cf.dir/build.make:109: src/mlpack/bindings/julia/mlpack/build/bin/generate_jl_cf] Error 1
[17:30:48] make[2]: Leaving directory '/workspace/srcdir/mlpack-3.3.0-a1/build'
[17:30:48] make[1]: *** [CMakeFiles/Makefile2:1797: src/mlpack/methods/cf/CMakeFiles/generate_jl_cf.dir/all] Error 2
[17:30:49] /opt/powerpc64le-linux-gnu/powerpc64le-linux-gnu/sys-root//usr/local/lib/libopenblas64_.so.0: undefined reference to `GOMP_parallel@GOMP_4.0'
[17:30:49] collect2: error: ld returned 1 exit status
[17:30:49] make[2]: *** [src/mlpack/methods/gmm/CMakeFiles/generate_jl_gmm_generate.dir/build.make:109: src/mlpack/bindings/julia/mlpack/build/bin/generate_jl_gmm_generate] Error 1
[17:30:49] make[2]: Leaving directory '/workspace/srcdir/mlpack-3.3.0-a1/build'
[17:30:49] make[1]: *** [CMakeFiles/Makefile2:2580: src/mlpack/methods/gmm/CMakeFiles/generate_jl_gmm_generate.dir/all] Error 2
```
you need to compile with GCC v5: those symbols are defined only in the libgomp library that comes with that version of the compiler:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"5")
```

This happens because OpenBLAS is [currently compiled](https://github.com/JuliaPackaging/Yggdrasil/blob/25683c75286048c81f37c87f111e23a1016c7c52/O/OpenBLAS/OpenBLAS%400.3.7/build_tarballs.jl#L16) with GCC v6.  It looks like this libgomp is incompatible with the one with GCC v4

Reply:
Thanks, I really appreciate that. :+1: Let's see how it does... :)

Reply:
For the record, all these issues are exactly the same we had with Sundials in #421.